app/assets/javascripts/host_edit.js

```diff
@@ -793,7 +795,7 @@ function interface_subnet_selected(element, ip_field, skip_mac) {
   preserve_selected_options($(element));
 
   var subnet_id = $(element).val();
```

Comment:
I'd check if it's disabled before even getting the value.
```suggestion
  if ($element.attr('disabled')) return;
  var subnet_id = $(element).val();
```